src/test/java/org/jenkinsci/plugins/workflow/cps/global/GrapeTest.java

```diff
@@ -0,0 +1,174 @@
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+@Issue("JENKINS-26192")
+public class GrapeTest {
+
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+    @Rule public RestartableJenkinsRule story = new RestartableJenkinsRule();
+    @Inject private WorkflowLibRepository repo;
+
+    @Test public void useBinary() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                FileUtils.write(new File(repo.workspace, "src/pkg/Lists.groovy"),
+                    "package pkg\n" +
+                    "@Grab('commons-primitives:commons-primitives:1.0')\n" +
+                    "import org.apache.commons.collections.primitives.ArrayIntList\n" +
+                    "static def arrayInt(script) {\n" +
+                    "  script.semaphore 'wait'\n" +
+                    "  new ArrayIntList()\n" +
+                    "}");
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("echo(/got ${pkg.Lists.arrayInt(this)}/)", true));
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("wait/1", b);
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                WorkflowRun b = p.getBuildByNumber(1);
+                SemaphoreStep.success("wait/1", null);
+                story.j.assertBuildStatusSuccess(story.j.waitForCompletion(b));
+                story.j.assertLogContains("got []", b);
+            }
+        });
+    }
+
+    @Test public void var() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                FileUtils.write(new File(repo.workspace, "vars/one.groovy"),
+                    "@Grab('org.apache.commons:commons-math3:3.4.1')\n" +
+                    "import org.apache.commons.math3.complex.ComplexField\n" +
+                    "def call() {ComplexField.instance.one}");
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("echo(/${one()} + ${one()} = ${one().add(one())}/)", true));
+                story.j.assertLogContains("(1.0, 0.0) + (1.0, 0.0) = (2.0, 0.0)", story.j.buildAndAssertSuccess(p));
+            }
+        });
+    }
+
+    // TODO test transitive dependencies; need to find something in Central which has a dependency not in this plugin’s test classpath and which could be used easily from a script
```

Comment:
Suggestions welcome. I spent considerable time looking for a usable example, without success.